sn_networking/src/record_store.rs

```diff
@@ -1461,120 +1460,140 @@ mod tests {
-                            }
-                            Err(err) => {
-                                panic!("Can't find close range of {name:?} with error {err:?}")
-                            }
-                        };
-
-                        let payee = pick_cheapest_payee(&peers_in_close, &peers);
-
-                        for peer in peers_in_replicate_range.iter() {
-                            let (close_records_stored, nanos_earnt, received_payment_count) =
-                                peers.entry(*peer).or_insert((0, 0, 0));
-                            if *peer == payee {
-                                let cost = calculate_cost_for_records(&QuotingMetrics {
-                                    close_records_stored: *close_records_stored,
-                                    max_records: MAX_RECORDS_COUNT,
-                                    received_payment_count: *received_payment_count,
-                                    live_time: 0,
-                                });
-                                *nanos_earnt += cost;
-                                *received_payment_count += 1;
-                            }
-                            *close_records_stored += 1;
+            let chunk_results: Vec<_> = (0..num_of_chunks_per_itr)
+                .into_par_iter()
+                .map(|_| {
+                    let name = xor_name::rand::random();
+                    let chunk_address = NetworkAddress::from_chunk_address(ChunkAddress::new(name));
+
+                    let chunk_distance_to_sorting = sorting_target_address.distance(&chunk_address);
+                    // Binary search to find the insertion point for the chunk
+                    let partition_point = peers.partition_point(|peer| {
+                        sorting_target_address.distance(&peer.address) < chunk_distance_to_sorting
+                    });
+
+                    // Collect close_group_size closest peers
+                    let mut close_group = Vec::with_capacity(close_group_size);
+                    let mut left = partition_point;
+                    let mut right = partition_point;
+
+                    while close_group.len() < close_group_size && (left > 0 || right < peers_len) {
+                        if left > 0 {
+                            left -= 1;
+                            close_group.push(left);
+                        }
+                        if right < peers_len && close_group.len() < close_group_size {
+                            close_group.push(right);
+                            right += 1;
                         }
                     }
-                    Err(err) => {
-                        panic!("Can't find replicate range of {name:?} with error {err:?}")
+
+                    // Sort the close group by distance to the chunk
+                    close_group.sort_by_key(|&i| chunk_address.distance(&peers[i].address));
```

Comment:
there is `while close_group.len() < close_group_size` in the above
which means this sort and truncate is not necessary.

unless you mean the above loop shall use something like `while close_group.len() < 5 * close_group_size` to reduce the chance of glitch that described above ?